FileSystemSimulator/IOSystem/Cylinder.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include "Track.h"
+class Cylinder
+{
+public:
+	static const int NUMBER_OF_TRACKS = 2;
+	Track** tracks;
+
+	Cylinder() {
+		tracks = new Track*[NUMBER_OF_TRACKS];
+		for (int i = 0; i < NUMBER_OF_TRACKS; i++) {
+			tracks[i] = new Track();
+		}
+	};
+
+	~Cylinder() {
+		delete[] tracks;
```

Comment:
you are deleting the array of poiners, but there is also a need to delete objects, which pointers refers to
```suggestion
                for (int i = 0; i < NUMBER_OF_TRACKS; i++)
                    delete tracks[i];
		delete[] tracks;
```